src/lambda_handler.py

```diff
@@ -34,54 +154,158 @@ def task_default(event):
+    """Publish the egress IP addresses in the given AWS accounts to an S3 bucket."""
+    result: Dict[str, Union[Optional[str], bool]] = {"message": None, "success": True}
 
-    character: str = event.get("character", "tux")
-    if character not in cowsay.characters.CHARS.keys():
-        error_msg = 'Character "%s" is not valid.'
-        logging.error(error_msg, character)
-        failed_task(result, error_msg % character)
-    else:
-        contents: str = event.get("contents", "Hello from AWS Lambda!")
-        logger.info(
-            'Creating output using "%s" with contents "%s"', character, contents
+    # An AWS-style filter definition to limit the queried regions
+    region_filters: List[Dict[str, Union[str, List[str]]]] = event.get(
+        "region_filters", []
+    )
+
+    # A list of dictionaries that define the files to be created and
+    # published.  When an IP is to be published, its associated
+    # application is compared to the app_regex field.  If it matches, it
+    # will be included in the associated filename.  The required keys in
+    # each dictionary are:
+    #   - "app_regex" (string): a regular expression that will be compared to
+    #       the application tag value to determine if the IP should be published
+    #   - "description" (string): a description of the file
+    #   - "filename" (string): the name of the file
+    #   - "static_ips" (list(string)): a list of CIDR blocks that will always
+    #       be included in the published file
+
+    class FileConfig(TypedDict):
+        """Define the type structure of the dictionary in the file_configs variable."""
+
+        app_regex: re.Pattern[str]
+        description: str
+        filename: str
+        # I'd prefer to define ip_set as Set[Union[IPv4Address, IPv6Address]],
+        # but ipaddress.collapse_addresses() uses the type variable "_N" while
+        # ip_network returns Union[IPv4Network, IPv6network] which causes
+        # the mypy pre-commit hook to throw this error:
+        #   Value of type variable "_N" of "collapse_addresses" cannot be
+        #   "Union[IPv4Network, IPv6Network]"
+        # My solution to this problem is to simply define ip_set as Set[Any].
+        # For a similar issue and discussion, see
+        # https://github.com/python/typeshed/issues/2080
+        ip_set: Set[Any]
+        static_ips: List[str]
+
+    file_configs: List[FileConfig] = event.get("file_configs", [])
+
+    # Header template for each file; the following variables are available
+    # within the template:
+    # {domain} - domain where the published files are located
+    # {filename} - name of the published file
+    # {timestamp} - timestamp when the file was published
+    # {description} - description of the published file
+    file_header: str = event.get(
+        "file_header",
+        "###\n# https://{domain}/{filename}\n# {timestamp}\n# {description}\n###\n",
```

Comment:
This is just a thought but would this be better represented as a `list(string)` of lines that get joined? That way it's a bit clearer what each piece of the header is when creating/reading for configuration. This could be future work if it seems reasonable.

Reply:
This is a nice idea, which I implemented in https://github.com/cisagov/publish-egress-ip-lambda/pull/4/commits/d27545d3653cce8d2c58987ba07baf42e2ae336d.

I now have to make corresponding changes over in https://github.com/cisagov/publish-egress-ip-terraform/pull/6, so let's not close this conversation until I have confirmed that is done.

Reply:
The corresponding Terraform change has been made over in https://github.com/cisagov/publish-egress-ip-terraform/pull/6/commits/d1c0a2906d3996c536eac8d198d23efc7175c303.